kythe/java/com/google/devtools/kythe/analyzers/base/KytheEntrySets.java

```diff
@@ -202,7 +202,11 @@ public EntrySet newAnchorAndEmit(VName fileVName, Span loc, Span snippet) {
 
   /** Emits and returns a DIAGNOSTIC node attached to no file. */
   public EntrySet emitDiagnostic(Diagnostic d) {
-    return emitDiagnostic(null, d);
+    VName fileVName =
+        useCompilationCorpusAsDefault
+            ? VName.newBuilder().setCorpus(compilationVName.getCorpus()).build()
+            : null;
+    return emitDiagnostic(fileVName, d);
```

Comment:
This incorrectly attaches the diagnostic to a file VName with an empty path.  The corpus needs to be set in the diagnostics builder

Reply:
Done. PTAL.  The diagnostics that are ending up without a corpus are those where the indexer is unable to find a symbol, so there's no file to attach the diagnostic to.

Reply:
Sorry for any confusion.

`emitDiagnostic(VName, Diagnostic)` will emit a `tagged` edge if the `VName` is non-null.  That's why this change is incorrect.  We will be adding a new `tagged` edge from a `VName` with only a corpus.

I believe this change is meant to only change the corpus on the `Diagnostic` itself which means the change should be located in `emitDiagnostic(VName, Diagnostic)` on both sides of the `fileVName == null` condition.

Reply:
Hmm, when you say to only change the corpus on the `Diagnostic` itself, where is it stored? The `Diagnostic` proto doesn't have any fields to store that AFAIK, only the `ResolvedDiagnostic` does.

Reply:
Can I just set the `CorpusPath` on the `NodeBuilder` in `emitDiagnostic(VName fileVName, Diagnostic d)`?

Reply:
Yeah, LGTM.